routemaster/tests/test_feeds.py

```diff
@@ -0,0 +1,72 @@
+import mock
+import pytest
+import httpretty
+
+from routemaster.feeds import Feed, FeedNotFetched, feeds_for_state_machine
+from routemaster.config import Feed as FeedConfig
+from routemaster.config import Gate, NoNextStates, StateMachine
+from routemaster.exit_conditions import ExitConditionProgram
+
+
+def test_feeds_for_state_machine():
+    state_machine = StateMachine(
+        name='example',
+        feeds=[
+            FeedConfig(name='test_feed', url='http://localhost/<label>'),
+        ],
+        states=[
+            Gate(
+                name='start',
+                triggers=[],
+                next_states=NoNextStates(),
+                exit_condition=ExitConditionProgram('false'),
+            ),
+        ]
+    )
+
+    feeds = feeds_for_state_machine(state_machine)
+
+    assert 'test_feed' in feeds
+    assert feeds['test_feed'].data is None
+    assert feeds['test_feed'].url == 'http://localhost/<label>'
+    assert feeds['test_feed'].state_machine == 'example'
+
+
+@httpretty.activate
+def test_fetch_only_once():
```

Comment:
I think this test is a duplicate of `test_only_loads_feed_once`

Reply:
Ish. This tests it directly on the feed, the other through a context. Given it's the context's responsibility to pre-warm I think it's worth having this test on both.